ShibbolethAuthPlugin.inc.php

```diff
@@ -30,22 +30,37 @@ class ShibbolethAuthPlugin extends GenericPlugin {
 	/** @var bool */
 	var $_globallyEnabled;
 
+	/** @var bool */
+	var $_singleContext;
+
 	/**
 	 * @copydoc Plugin::__construct()
 	 */
 	function __construct() {
 		parent::__construct();
 		$this->_contextId = $this->getCurrentContextId();
 		$this->_globallyEnabled = $this->getSetting(0, 'enabled');
+                $this->_singleContext = true;
```

Comment:
I wonder if this is needed, given that this is a "sitewide" plugin (c.f. [version.xml](https://github.com/pkp/shibboleth/blob/5a57107bcc3dc8059c480bb96b451333d71a19c7/version.xml#L19) and [ShibbolethAuthPlugin.inc.php](https://github.com/pkp/shibboleth/blob/5a57107bcc3dc8059c480bb96b451333d71a19c7/ShibbolethAuthPlugin.inc.php#L79-L84).  That is, you wouldn't want to enable it for just one journal in a site, right?  The fix for the single journal install might be easier and clearer as forcing `$_globallyEnabled` and ensuring settings are always written to `CONTEXT_SITE`.

Reply:
Oh ok, I'll give that a go and see. I followed the changes from [this ojs_ldap_plugin](https://github.com/shemgp/ojs_ldap_plugin/pull/3/commits/1579dfe6ff07f1fc57e27e4961b5f78023da36a4). I'll have a look at the documentation for `CONTEXT_SITE`.

Reply:
Interestingly, the ojs_ldap_plugin also describes itself as sitewide.

Generally, we pass a context id (that is, "which journal does this apply to?") into the constructor and functions, but since this is a site plugin, we could just ignore the passed in context id and force everything to the `CONTEXT_SITE` constant.

On the other hand, this then raises the question of what happens for folks who already have data assigned within a specific context?  We'd either need to migrate the data for them (probably a bad idea), or require that they re-do their settings after the upgrade (probably a slightly less bad idea).

Or could there actually be a use case out there where different journals would have different Shib configurations, or LDAP configurations?

Reply:
I personally cannot see the use for seperate journals having different authentication methods. The reason I added this change in the first place was for OJS instances that only had single journals. They had issue enabling Shibboleth because it wouldn't let them enable it on their journals, and because they only have one journal they don't have access to the site wide plugins.